### PR TITLE
My account - V1

### PR DESCRIPTION
**Pull request by eskil0312**

### CODE REVIEW

**Review by eskil0312 — COMMENTED** [review]
Will update fusion api when approved and published https://github.com/equinor/fusion-api/pull/255

File: src/components/core/Header/components/AccountManagerSideSheet/MenuComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import * as styles from './styles.less';
+import { UserMenuSectionItem, PersonDetails } from '@equinor/fusion';
+import { IconButton, ArrowBackIcon } from '@equinor/fusion-components';
+
+type MenuComponentProps = {
+    selectedMenuItem: UserMenuSectionItem;
+    onBackClick: () => void;
```
Comment (odinr) [inline src/components/core/Header/components/AccountManagerSideSheet/MenuComponent.tsx:8]: *tip* [VoidFunction](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.voidfunction.html)

File: src/components/people/PersonPositionList/PersonPositionCards.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    id: string;
+    positions: PersonPosition[];
+};
+
+type CardLinkProps = {
+    position: PersonPosition;
+    disableOrgLink: boolean;
+};
+
+const CardLink: React.FC<CardLinkProps> = ({ children, disableOrgLink, position }) => {
```
Comment (odinr) [inline src/components/people/PersonPositionList/PersonPositionCards.tsx:22]: think you need to wrap the `CardLinkProps` with `PropsWithChildren<CardLinkProps>`
it works, but guessing the linter will complain.
then simply return children without the fragment wrapper

File: src/components/people/PersonPositionList/PersonPositionCards.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                        id: position.project.id,
+                        positions: [position],
+                    });
+                    return prevPositionProject;
+                },
+                []
+            ),
+        [positions]
+    );
+    console.log(disableOrgLink)
```
Comment (odinr) [inline src/components/people/PersonPositionList/PersonPositionCards.tsx]: :)

File: src/components/people/PersonPositionList/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    positions.filter(
+        (position) =>
+            filterToDate.getTime() >= position.appliesFrom.getTime() &&
+            filterToDate.getTime() <= position.appliesTo.getTime()
+    );
+export const getPastPositions = (positions: PersonPosition[], filterToDate: Date) =>
+    positions.filter((position) => position.appliesTo.getTime() < filterToDate.getTime());
+
+export const getFuturePositions = (positions: PersonPosition[], filterToDate: Date) =>
+    positions.filter((position) => position.appliesFrom.getTime() > filterToDate.getTime());
```
Comment (odinr) [inline src/components/people/PersonPositionList/helpers.ts:16]: provide return type.

File: src/components/people/PersonRoleList/RoleItem.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                            active={isActive}
+                            disabled={!role.onDemandSupport}
+                            onChange={() => toggleRoleStatus(role)}
+                        />
+                    </div>
+                )}
+            </div>
+            <div className={styles.row}>
+                <div className={styles.small}>{role.name}</div>
+                {role.activeToUtc && showSwitch && !errorMessage && (
```
Comment (odinr) [inline src/components/people/PersonRoleList/RoleItem.tsx:83]: this works, but hard to read/understand, pref assing to a variable like: `const showExpireDate = role.activeToUtc && showSwitch && !errorMessage;` or into a memo

**Review by odinr — APPROVED** [review]
Looks good, only minor code styling

File: src/components/people/PersonPositionList/PersonPositionCards.tsx (hunk 2 above)
Comment (eskil0312) [reply src/components/people/PersonPositionList/PersonPositionCards.tsx:22]: After adding PropsWithChildren the linter still complains if I remove the fragment

**Pull request merged**